base/deprecated.jl

```diff
@@ -785,4 +785,89 @@ const _oldstyle_array_vcat_ = false
+    S = esc(S)
+    f = esc(f)
+    quote
+        @deprecate $f{T1<:$S,T2<:$S}(x::T1, y::AbstractArray{T2}) $f.(x,y)
+        @deprecate $f{T1<:$S,T2<:$S}(x::AbstractArray{T1}, y::T2) $f.(x,y)
+        @deprecate $f{T1<:$S,T2<:$S}(x::AbstractArray{T1}, y::AbstractArray{T2}) $f.(x,y)
+    end
+end
+
+# Deprecate @vectorize_1arg-vectorized functions from...
+for f in (
+        :sinpi, :cospi, :sinc, :cosc, # base/special/trig.jl
+        :log, :log1p, # base/special/log.jl
+        :gamma, :lfact, :digamma, :trigamma, :zeta, :eta,# base/special/gamma.jl
+        :erfcx, :erfi, :dawson, # base/special/erf.jl
+        :airyprime, :airyai, :airyaiprime, :airybi, :airybiprime, # base/special/bessel.jl
+            :airy, :airyx, :besselj0, :besselj1, :bessely0, :bessely1, # base/special/bessel.jl
+        :cbrt, :sinh, :cosh, :tanh, :atan, :asinh, :exp, :erf, :erfc, :exp2, # base/math.jl
+            :expm1, :exp10, :sin, :cos, :tan, :asin, :acos, :acosh, :atanh, # base/math.jl
+            #=:log,=# :log2, :log10, :lgamma, #=:log1p,=# :sqrt,            # base/math.jl
+        :abs, :abs2, :angle, :isnan, :isinf, :isfinite, # base/floatfuncs.jl
+        :acos_fast, :acosh_fast, :angle_fast, :asin_fast, :asinh_fast, # base/fastmath.jl
+            :atan_fast, :atanh_fast, :cbrt_fast, :cis_fast, :cos_fast, # base/fastmath.jl
+            :cosh_fast, :exp10_fast, :exp2_fast, :exp_fast, :expm1_fast, # base/fastmath.jl
+            :lgamma_fast, :log10_fast, :log1p_fast, :log2_fast, :log_fast, # base/fastmath.jl
+            :sin_fast, :sinh_fast, :sqrt_fast, :tan_fast, :tanh_fast, # base/fastmath.jl
+        :cis, # base/complex.jl
+        )
+    @eval @dep_vectorize_1arg Number $f
+end
+for f in (
+        :sind, :cosd, :tand, :asind, :acosd, :atand, :asecd, :acscd, :acotd, # base/special/trig.jl
+        :invdigamma, # base/special/gamma.jl
+        :erfinc, :erfcinv, # base/special/erf.jl
+        :rad2deg, :deg2rad, :exponent, :significand, # base/math.jl
+        :trunc, :floor, :ceil, :round, # base/floatfuncs.jl
+        :unix2datetime, :rata2datetime, :julian2datetime, # base/dates/conversions.jl
+        )
+    @eval @dep_vectorize_1arg Real $f
+end
+# base/complex.jl
+@dep_vectorize_1arg Complex round
+@dep_vectorize_1arg Complex float
+for f in (
+        :year, :month, :day, :week, :dayofmonth, :yearmonth, :monthday, :yearmonthday, # base/dates/accessors.jl
+        :firstdayofweek, :lastdayofweek, :firstdayofmonth, # base/dates/adjusters.jl
+            :lastdayofmonth, :firstdayofyear, :lastdayofyear, # base/dates/adjusters.jl
+            :firstdayofquarter, :lastdayofquarter, # base/dates/adjusters.jl
+        :dayname, :dayabbr, :dayofweek, :dayofweekofmonth, # base/dates/query.jl
+            :daysofweekinmonth, :monthname, :monthabbr, :daysinmonth, # base/dates/query.jl
+            :isleapyear, :dayofyear, :daysinyear, :quarterofyear, :dayofquarter, # base/dates/query.jl
+    )
+    @eval @dep_vectorize_1arg Dates.TimeType $f
```

Comment:
Seems like this depends on #16966 — i.e. we need `broadcast` to recognize `TimeType` as a "scalar".